daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -344,6 +346,16 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
    */
   def readVariable(vrd: VariableRuntimeData, referringContext: ThrowsSDE, maybeState: Maybe[ParseOrUnparseState]): DataValuePrimitive = {
     val varQName = vrd.globalQName
+    if (maybeState.isDefined) {
+      vrd.direction match {
+        case VariableDirection.ParseOnly if (!maybeState.get.isInstanceOf[PState]) =>
+          maybeState.get.SDW("Attempting to read variable %s which is marked as parseOnly during unparsing".format(varQName))
+        case VariableDirection.UnparseOnly if (!maybeState.get.isInstanceOf[UState]) =>
+          maybeState.get.SDW("Attempting to read variable %s which is marked as unparseOnly during parsing".format(varQName))
+        case _ => // Do nothing
```

Comment:
Should these be SDEs?

Reply:
This is the one thing I'm not sure about.  It's an easy change to make, but given that there is no spec for this feature I wasn't sure what we would want to happen here.

Reply:
Definitely make them SDE. We can always weaken them to SDW in the future if we find SDE too restrictive. But I think SDE is right.